Bluejay/Bluejay/Bluejay.swift

```diff
@@ -663,11 +677,18 @@ extension Bluejay: CBCentralManagerDelegate {
         UIApplication.shared.endBackgroundTask(backgroundTask)
     }
     
+    /**
+     Figure out whether this is any kinds of disconnection when Bluejay is still connecting, an expected disconnection from a connected state, or an unexpected disconnection from a connected state. An unexpected disconnection from a connected state, such as walking out of range or restarting the Bluetooth device, is the only case where `shouldAutoReconnect` should be set to true.
```

Comment:
This is another example where the doc comment looks more like the narrative of what we're currently doing in the implementation, rather than what the public interface for this is useful for and how it's to be used. Skimming this it's not clear to me why this is public or how it's intended to be used?